lib/actions/mcubootTargetActions.js

```diff
@@ -0,0 +1,216 @@
+ *    engineered, decompiled, modified and/or disassembled.
+ *
+ * THIS SOFTWARE IS PROVIDED BY NORDIC SEMICONDUCTOR ASA "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, NONINFRINGEMENT, AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NORDIC SEMICONDUCTOR ASA OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/* eslint-disable import/no-cycle */
+
+import { logger } from 'nrfconnect/core';
+import nrfjprog from 'pc-nrfjprog-js';
+
+import { CommunicationType } from '../util/devices';
+import * as targetActions from './targetActions';
+import * as warningActions from './warningActions';
+
+export const MCUBOOT_KNOWN = 'MCUBOOT_KNOWN';
+export const MCUBOOT_PORT_KNOWN = 'MCUBOOT_PORT_KNOWN';
+export const MCUBOOT_PROCESS_UPDATE = 'MCUBOOT_PROCESS_UPDATE';
+export const MCUBOOT_WRITING_CLOSE = 'MCUBOOT_WRITING_CLOSE';
+export const MCUBOOT_WRITING_END = 'MCUBOOT_WRITING_END';
+export const MCUBOOT_WRITING_FAIL = 'MCUBOOT_WRITING_FAIL';
+export const MCUBOOT_WRITING_READY = 'MCUBOOT_WRITING_READY';
+export const MCUBOOT_WRITING_START = 'MCUBOOT_WRITING_START';
+export const MCUBOOT_WRITING_SUCCEED = 'MCUBOOT_WRITING_SUCCEED';
+
+export const MCUBOOT_DFU_NOT_STARTED = 'Not started.';
+export const MCUBOOT_DFU_STARTING = 'Starting...';
+
+export const mcubootKnownAction = isMcuboot => ({
+    type: MCUBOOT_KNOWN,
+    isMcuboot,
+});
+
+export const portKnownAction = port => ({
+    type: MCUBOOT_PORT_KNOWN,
+    port,
+});
+
+export const processUpdateAction = (message, percentage, duration) => ({
+    type: MCUBOOT_PROCESS_UPDATE,
+    message,
+    percentage,
+    duration,
+});
+
+export const writingReadyAction = fileName => ({
+    type: MCUBOOT_WRITING_READY,
+    fileName,
+});
+
+export const writingCloseAction = () => ({
+    type: MCUBOOT_WRITING_CLOSE,
+});
+
+export const writingStartAction = () => ({
+
+    type: MCUBOOT_WRITING_START,
+});
+
+export const writingEndAction = () => ({
+    type: MCUBOOT_WRITING_END,
+});
+
+export const writingSucceedAction = () => ({
+    type: MCUBOOT_WRITING_SUCCEED,
+});
+
+export const writingFailAction = errorMsg => ({
+    type: MCUBOOT_WRITING_FAIL,
+    errorMsg,
+});
+
+export const openDevice = selectedDevice => dispatch => {
+    dispatch(targetActions.targetTypeKnownAction(CommunicationType.MCU, false));
+    dispatch(mcubootKnownAction(true));
+    dispatch(portKnownAction(selectedDevice['serialport.1'].comName));
```

Comment:
This is too fragile. Find the correct serialport by verifing the pnpId which will be platform dependent.
Also verify what serialport object looks like on macOS to be sure, there pnpId was not working earlier.
https://github.com/NordicSemiconductor/pc-nrfconnect-linkmonitor/blob/master/index.jsx#L58

Reply:
Good suggestion 👍
Updated with suggestion but still need to check and test on other PCs.